fe/fe-core/src/main/java/com/starrocks/transaction/DatabaseTransactionMgr.java

```diff
@@ -700,6 +700,10 @@ public boolean canTxnFinished(TransactionState txn, Set<Long> errReplicas, Set<L
                                             && (unfinishedBackends == null
                                             || !unfinishedBackends.contains(replica.getBackendId()))) {
                                         ++successHealthyReplicaNum;
+                                    // replica report version has greater cur transaction commit version
+                                    // This can happen when the BE publish succeeds but fails to send a response to FE
+                                    } else if (replica.getVersion() >= partitionCommitInfo.getVersion()) {
```

Comment:
There has a same version check following.
I think it is better to move the check to front, rather than adding new one.

Reply:
the difference is `!errReplicas.contains(replica.getId())` condition, if it not true, it will execute `errReplicas.remove(replica.getId())`, otherwise all the way need to judge this condition

Reply:
if replica version >= commit info version, the replica is healthy and is reasonable to remove it from `errReplicas`.